src/compilers/approx.lisp

```
@@ -763,7 +731,8 @@ NOTE: This routine degenerates to an optimal 2Q compiler when *ENABLE-APPROXIMAT
                       (sandwich-with-local-gates center-circuit a d b q1 q0)
                     (push (cons (* circuit-cost fidelity) sandwiched-circuit)
                           candidate-pairs)))
-              (compiler-does-not-apply () nil))))
+              (compiler-does-not-apply () nil)
+              (diagonalizer-not-found () (give-up-compilation)))))
```

@ecpeterson What do you think here?

---

This isn't fully unreasonable, but it doesn't have the behavior I was imagining: `give-up-compilation` often signals an expected error that gets gracefully handled by the caller. `diagonalizer-not-found` is an unexpected error, and I'd like the caller to have the option to do something about it—including proceed "as normal", as if `give-up-compilation` had been emitted instead—but I don't like the idea of the unexpected error getting silently and automatically converted into the meant-to-be-recovered-from error.

But, again, it's ultimately not that bad in context. If this routine flags `give-up-compilation` to its usual caller, what will happen in 99% of cases is catastrophic failure shortly thereafter. So, stack traces will be murkier, but we won't start emitting bad output.

---

So should we remove this handler, and just let the error bubble up?

---

Yeah, I suppose so. You could consider trapping it in apply-translation-compilers with the same option to restart with the next compiler, but it’d be fine to not bother.